composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/DefaultGameComponent.kt

```diff
@@ -21,10 +22,10 @@ internal class DefaultGameComponent(
     componentContext: ComponentContext,
     storeFactory: StoreFactory,
     settings: GameSettings,
-    coroutineContext: CoroutineContext
+    mainCoroutineContext: CoroutineContext
 ) : GameComponent, ComponentContext by componentContext {
 
-    private val scope = coroutineScope(context = coroutineContext)
+    private val scope = coroutineScope(context = SupervisorJob() + mainCoroutineContext)
```

Comment:
I'm not sure if it's related, but usually it's the opposite: `mainCoroutineContext + SupervisorJob()`.

Reply:
Hmm, okay, i make changes that you talking about, somewhere I saw how I'm declare it.
Is it just for best eye intercept to show that this context has also parent context, that coming first, before +?
Or it has some logic behind it, for coroutine work?

Reply:
I think it's the [docs](https://kotlinlang.org/docs/exception-handling.html#supervision). Also this way is [used](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:lifecycle/lifecycle-viewmodel/src/commonMain/kotlin/androidx/lifecycle/viewmodel/internal/CloseableCoroutineScope.kt;l=60) in android ViewModel.